cargo-near-build/src/near/build/mod.rs

```diff
@@ -165,7 +171,10 @@ pub fn run(args: Opts) -> eyre::Result<CompilationArtifact> {
 
     wasm_artifact.path = {
         let prev_artifact_path = wasm_artifact.path;
-        let target_path = out_dir.join(prev_artifact_path.file_name().expect("has filename"));
+
+        // NOTE important!: the way the output path for wasm is resolved now cannot change,
+        // see more detail on [CrateMetadata::get_legacy_cargo_near_output_path]
+        let target_path = crate_metadata.get_legacy_cargo_near_output_path(args.out_dir.clone())?;
```

Comment:
this change doesn't change the result, but only tries to highlight, that the implementation
should better stay backwards compatible